tests/unit/yaml_utils/test_yaml_utils.py

```diff
@@ -15,39 +15,48 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import io
-import os
+import pytest
 
-from testtools.matchers import Equals
 
 from snapcraft import yaml_utils
 from snapcraft.yaml_utils import YamlValidationError
-from tests import unit
 
 
-class YamlLoadTests(unit.TestCase):
-    def test_load_yaml_file(self):
-        path = os.path.join(self.path, "test.yaml")
-        with open(path, "w") as f:
-            f.write("foo: bar")
+def test_load_yaml_file(caplog, tmp_path):
+    yaml_path = tmp_path / "test.yaml"
+    yaml_path.write_text("foo: bar")
 
-        cfg = yaml_utils.load_yaml_file(path)
+    cfg = yaml_utils.load_yaml_file(str(yaml_path))
 
-        self.assertThat(cfg, Equals({"foo": "bar"}))
+    assert cfg == {"foo": "bar"}
+    assert [r.message for r in caplog.records] == []
 
-    def test_load_yaml_file_bad_yaml(self):
-        path = os.path.join(self.path, "test.yaml")
-        with open(path, "w") as f:
-            # Note the missing newlines...
-            f.write("foo: bar")
-            f.write("a-b-c: xyz")
 
-        self.assertRaises(YamlValidationError, yaml_utils.load_yaml_file, path)
+def test_load_yaml_file_bad_yaml(caplog, tmp_path):
+    yaml_path = tmp_path / "test.yaml"
+    yaml_path.write_text("foo: bara-b-c: xyz\n")
 
+    with pytest.raises(YamlValidationError):
+        yaml_utils.load_yaml_file(str(yaml_path))
 
-class OctIntTest(unit.TestCase):
-    def test_octint_dump(self):
-        output = io.StringIO()
-        yaml_utils.dump(dict(number=yaml_utils.OctInt(8)), stream=output)
-        output.seek(0)
+    assert [r.message for r in caplog.records] == []
 
-        self.assertThat(output.read().strip(), Equals("number: 0010"))
+
+def test_load_yaml_file_with_duplicate_warnings(caplog, tmp_path):
```

Comment:
can you add a nest with nest duplicates?

Reply:
Adding a test with the following scenario, is this what you mean? (In this case outer `foo` must not clash with inner `foo`, and inner `baz` must be detected)
```
foo: bar
nest:
  foo: bar2
  baz: bar3
  baz: bar4
```